### DISCUSSION TITLE
Try out ESLint in Next.js (experimental)

### DISCUSSION THREAD

**Discussion by housseindjirdeh** [vercel/next.js#25008] [Ideas]
#24900 covers all the steps that are currently underway to land out-of-the-box ESLint support in Next.js and Create Next App. As of now, ESLint can be enabled to run during builds in Next.js via an experimental flag in case you would like to try it out sooner.

# Steps
 
1. In `package.json`, update your `next` dependency to the latest canary version:

```
  "dependencies": {
    "next": "10.2.1-canary.5",
    //....
  }
 ```

Re-install the dependency if needed: `rm -rf node_modules/next && yarn install --force`

2. Create (or update) the `next.config.js` file in the root of your project and enable the experimental `eslint` flag:

```js
module.exports = {
  experimental: {
    eslint: true,
  },
};
```

3. Create an empty `.eslintrc` file in the root of your project (if you don't have one included already):

```bash
touch .eslintrc
```

4. Install required dependencies for ESLint, including the latest version of Next.js' shareable config:

```
yarn add --dev eslint eslint-config-next@10.2.1-canary.5
```

5. Run a build with `npm run build` or `yarn build`. If `.eslintrc` was empty, Next.js will automatically populate it with a default config.

Now ESLint will run on every build 🎉

**Comment by leerob** [comment]
Worked really well! 🚀 

https://github.com/leerob/leerob.io/pull/318

**Reply by justirva09** [reply]
Hi Maintainers, how to get it to work with code-prettier?

**Comment by stefanprobst** [comment]
what do you think about using `eslint-import-resolver-typescript` instead of `eslint-import-resolver-node` for typescript files in `eslint-config-next`? the first supports absolute import `paths` defined in `tsconfig`, the second does not.

```json
{
  "settings": {
    "import/resolver": {
      "typescript": {
        "extensions": [".ts", ".tsx"],
        "alwaysTryTypes": true
      }
    }
  }
}
```

**Reply by stefanprobst** [reply]
@housseindjirdeh what's your opinion on the above?
